tensorboard/plugins/histogram/histograms_plugin.py

```diff
@@ -120,11 +126,37 @@ def index_impl(self):
         summary_metadata = self._multiplexer.SummaryMetadata(run, tag)
         result[run][tag] = {'displayName': summary_metadata.display_name,
                             'description': plugin_util.markdown_to_safe_html(
-                                summary_metadata.summary_description)}
+                              summary_metadata.summary_description)}
 
     return result
 
-  def histograms_impl(self, tag, run, downsample_to=None):
+  def _get_csv_response(self, events):
+    """
+    Returns a string response with all the events formatted in the csv format
+    :param events:
+    :return:
+    """
+    csv_events = []
+    # Convert the events in a way that we can sensibly export
+    # them as csv. Therefore, we split the start, end, value of
+    # each bin by semicolon.
+    for e in events:
+      csv_events.append(
```

Comment:
Optional but simpler to just create the csv.writer at the top and then call writer.writerow() instead of building up the intermediate csv_events list.